src/services/NodeMonitor.ts

```diff
@@ -81,31 +90,29 @@ export class NodeMonitor {
 	};
 
 	private getNodeList = async (): Promise<any> => {
-		logger.info(`Getting node list`);
+		logger.info(`[getNodeList] Getting node list...`);
+		const startTime = new Date().getTime();
 
 		// Fetch node list from config nodes
+		logger.info(`[getNodeList] Initial node list: ${symbol.NODES.join(', ')}`);
 		for (const nodeUrl of symbol.NODES) {
 			const peers = await this.fetchNodePeersByURL(nodeUrl);
 
 			this.addNodesToList(peers);
 		}
 
-		// Nested fetch node list from current nodeList[]
-		const nodeListPromises = this.nodeList.map(async (node) => {
-			if (isAPIRole(node.roles)) {
-				const hostUrl = await ApiNodeService.buildHostUrl(node.host);
-
-				return this.fetchNodePeersByURL(hostUrl);
-			}
+		// Fetch node list from database
+		const nodesFromDb = (await DataBase.getNodeList().then((nodes) => nodes.map((n) => n.toJSON()))) || [];
 
-			return [];
-		});
+		logger.info(`[getNodeList] Nodes count from DB: ${nodesFromDb.length}`);
+		// adding the nodes from DB to the node list
+		this.addNodesToList(nodesFromDb);
```

Comment:
Seems like dead / disconnected nodes can stay in the list?

Reply:
It's a known issue, explained in the PR description, addressed in issue https://github.com/symbol/statistics-service/issues/122